source/NVDAObjects/window/edit.py

```diff
@@ -274,8 +277,24 @@ def _getFormatFieldAndOffsets(self,offset,formatConfig,calculateOffsets=True):
 			    formatField["text-position"] = TextPosition.BASELINE
 		if formatConfig["reportColor"]:
 			if charFormat is None: charFormat=self._getCharFormat(offset)
-			formatField["color"]=colors.RGB.fromCOLORREF(charFormat.crTextColor) if not charFormat.dwEffects&CFE_AUTOCOLOR else _("default color")
-			formatField["background-color"]=colors.RGB.fromCOLORREF(charFormat.crBackColor) if not charFormat.dwEffects&CFE_AUTOBACKCOLOR else _("default color")
+			if charFormat.dwEffects & CFE_AUTOCOLOR:

```

Comment:
could these code additions be factored out to a helper function? It would be good to simplify this function where possible. It may be worth adding the code complexity warning to this function (and the others changed in this file if relevant).

Reply:
Fixed in c118f3be335df4638ff303edaebd9851bc17ba6c, i.e. processing related to color reporting factored out of 2 modified function and added code complexity noqa for these same 2 functions.